examples/https.rs

```diff
@@ -0,0 +1,23 @@
+//! Simple HTTPS GET
+//!
+//! This example is a Rust adaptation of the [C example of the same
+//! name](https://curl.se/libcurl/c/https.html).
+
+extern crate curl;
+
+use curl::easy::Easy;
+use std::io::{stdout, Write};
+
+fn main() -> Result<(), curl::Error> {
+    let mut curl = Easy::new();
+
+    curl.url("https://example.com/")?;
+    curl.write_function(|data| {
+        stdout().write_all(data).unwrap();
+        Ok(data.len())
+    })?;
+
+    curl.perform()?;
```

Comment:
It would be nice to also verify the returned body bytes here, to guard against truncation or other corruption problems. Also, FWIW I like using httpbin.org for these sorts of public-Internet tests, in part because you can ask it to generate something with a known body. For instance, http://httpbin.org/#/Dynamic_data.

Reply:
I mostly added this as just a quick way of running an HTTPS request manually, it isn't meant to be considered a _test_. It is a direct translation of an upstream C example, which I can see being helpful and something I'll probably add more of in the future.

It would be nice to add an HTTPS check to the tests just for validation though.